react/features/welcome/components/WelcomePage.native.js

```diff
@@ -65,47 +97,164 @@ class WelcomePage extends AbstractWelcomePage {
-                        placeholder = { t('welcomepage.roomname') }
-                        placeholderTextColor = { PLACEHOLDER_TEXT_COLOR }
-                        returnKeyType = { 'go' }
-                        style = { styles.textInput }
-                        underlineColorAndroid = 'transparent'
-                        value = { this.state.room } />
-                    <View style = { styles.buttonRow }>
-                        <TouchableHighlight
-                            accessibilityLabel = { 'Tap for Settings.' }
-                            onPress = { this._onSettingsOpen }
-                            style = { [ styles.button, styles.settingsButton ] }
-                            underlayColor = { ColorPalette.white }>
+                <View style = { PlatformElements.page }>
+                    <Header style = { styles.header }>
+                        <TouchableOpacity onPress = { this._onShowSideBar } >
                             <Icon
-                                name = 'settings'
-                                style = { styles.settingsIcon } />
-                        </TouchableHighlight>
+                                name = 'menu'
+                                style = { PlatformElements.headerButton } />
+                        </TouchableOpacity>
+                        <View style = { styles.audioVideoSwitchContainer }>
+                            <Text style = { PlatformElements.headerText } >
+                                { t('welcomepage.videoEnabledLabel') }
+                            </Text>
+                            <Switch
+                                onTintColor = { SWITCH_UNDER_COLOR }
+                                onValueChange = { this._onStartAudioOnlyChange }
+                                style = { styles.audioVideoSwitch }
+                                thumbTintColor = { SWITCH_THUMB_COLOR }
+                                value = { _profile.startAudioOnly } />
+                            <Text style = { PlatformElements.headerText } >
+                                { t('welcomepage.audioOnlyLabel') }
+                            </Text>
+                        </View>
+                    </Header>
+                    <SafeAreaView style = { styles.roomContainer } >
+                        <View style = { PlatformElements.paddedView } >
+                            <TextInput
+                                accessibilityLabel = { 'Input room name.' }
+                                autoCapitalize = 'none'
+                                autoComplete = { false }
+                                autoCorrect = { false }
+                                autoFocus = { false }
+                                onBlur = { this._onFieldFocusChange(false) }
+                                onChangeText = { this._onRoomChange }
+                                onFocus = { this._onFieldFocusChange(true) }
```

Comment:
Does TextInput not pass back if it is blurred or not? That would remove the need to create new functions on re-render.

Reply:
I couldn't find any reference to that in the documentation. Do you know any more details about it?

Reply:
I have no idea why I thought they would pass back something useful. I guess what I was thinking about was a way to re-use functions instead of creating new functions on each render. Generally I'm not in favor of the pattern of passing in new functions to an element if that's easy to avoid as an example of avoiding potentially unnecessary re-rendering. So even having this._onFocus and this._onBlur which call into this._genericFocusChangeHangle(bool) jives better with me.

Reply:
Yeah, that can work, thanks, I'll change it